assembly/src/parsers/mod.rs

```diff
@@ -142,26 +147,47 @@ pub struct ProcedureAst {
 }
 
 impl Serializable for ProcedureAst {
-    /// Writes byte representation of the `ProcedureAst` into the provided `ByteWriter` struct.
-    fn write_into(&self, target: &mut ByteWriter) -> Result<(), SerializationError> {
-        self.name.write_into(target)?;
-        self.docs.write_into(target)?;
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        self.name.write_into(target);
+        match &self.docs {
+            Some(t) => {
+                target.write_bool(true);
+                target.write_u64(t.len() as u64);
+                target.write_bytes(t.as_bytes());
+            }
+            None => {
+                target.write_bool(false);
+            }
+        }
         target.write_bool(self.is_export);
         target.write_u16(self.num_locals);
-        self.body.write_into(target)?;
-        Ok(())
+        target.write_u64(self.body.len() as u64);
+        self.body.write_into(target);
```

Comment:
I think `u64` both for docs and for body length is too much - let's use `u16` and add TODOs to enforce these restrictions via the constructor.